src/cdk/table/table.ts

```diff
@@ -474,6 +505,12 @@ export class CdkTable<T> implements AfterContentChecked, CollectionViewer, OnDes
     this._dataDiffer = this._differs.find([]).create((_i: number, dataRow: RenderRow<T>) => {
       return this.trackBy ? this.trackBy(dataRow.dataIndex, dataRow.data) : dataRow;
     });
+
+    // Table cell dimensions may change after resizing the window. Signal the sticky styler to
+    // refresh its cache of cell widths the next time sticky styles are updated.
+    this._viewportRuler.change().subscribe(() => {
```

Comment:
In general, any `subscribe` should be matched with a corresponding `unsubscribe` somewhere else, otherwise it'll get stuck in memory. 

You can use this pipe to handle it: `.pipe(takeUntil(this._onDestroy))`, just like we do with the dataStream in `_observeRenderChanges`